DOCUMENTATION.md

```diff
@@ -37,61 +38,114 @@ The output tensor `ys` will have shape `(t_size, batch_size, state_size)`, and c
+If training neural SDEs _with_ the adjoint method, then specifically Stratonovich SDEs and `reversible_heun` come strongly recommended; [see the notes on adjoints](#adjoints).
 
-Using the adjoint method will reduce memory usage, but takes (much) longer to compute. In addition, because the backward SDE is solved numerically, there will be a certain amount of numerical error, which can be nontrivial for large step sizes.
+## Adjoints
 
-If using the adjoint method, then we recommend using Stratonovich SDEs if possible, as these are faster to calculate the adjoint SDE for. We also recommend using adaptive step sizes on both the forward and backward passes, as this will help to reduce the numerical error.
+The SDE solve may be backpropagated through. This may be done either by backpropagating through the internal operations of the solver (when using `sdeint`), or via the _adjoint method_ [\[2\]](https://arxiv.org/pdf/2001.01328.pdf), which solves another "adjoint SDE" backwards in time to calculate the gradients (when using `sdeint_adjoint`).
+
+Using the adjoint method will reduce memory usage, but takes longer to compute.
 
 `sdeint_adjoint` supports all the arguments and keyword arguments that `sdeint` does, as well as the following additional keyword arguments: 
-- `adjoint_method`, `adjoint_adaptive`, `adjoint_rtol`, `adjoint_atol`: as for the forward pass.  
 
+- `adjoint_method`, `adjoint_adaptive`, `adjoint_rtol`, `adjoint_atol`: as for the forward pass.  
 - `adjoint_params`: the tensors to calculate gradients with respect to. If not passed, defaults to `sde.parameters()`.
 
-Double backward is supported through the adjoint method, and is calculated using an adjoint-of-adjoint SDE, with the same method, tolerances, etc. as for the adjoint SDE. Note that the numerical errors can easily grow very large for adjoint-of-adjoint SDEs.
+Double backward is supported through the adjoint method, and is calculated using an adjoint-of-adjoint SDE, with the same method, tolerances, etc. as for the adjoint SDE. Note that the numerical errors can easily grow large for adjoint-of-adjoint SDEs.
+
+#### Advice on adjoint methods
+
+Use Stratonovich SDEs if possible, as these have a computationally cheaper adjoint SDE than Ito SDEs.
+
+Solving the adjoint SDE implies making some numerical error. If not managed carefully then this can make training more difficult, as the gradients calculated will be less accurate. Whilst the issue can just be ignored -- it's usually not a deal-breaker -- there are two main options available for managing it:
+
+- The usual best approach is to use `method="reversible_heun"` and `adjoint_method="adjoint_reversible_heun"`, as introduced in [\[1\]](https://arxiv.org/abs/2105.13493). These are a special pair of solvers which when used in conjunction have almost zero numerical error, as the adjoint solver carefully reconstructs the numerical trajectory taken by the forward solver.
```

Comment:
Ideally, `adjoint_reversible_heun` should be automatically chosen when forward is using `reversible_heun`. This currently seems to break with adaptive solvers.